test/utils.ts

```diff
@@ -41,6 +41,10 @@ export function makeCompilationEnvironment(options: Record<string, any>): Compil
     return new CompilationEnvironment(compilerProps, compilationQueue, options.doCache);
 }
 
+export function makeFake<T>(val: Partial<T>): T {
+    return val as T;
+}
+
```

Comment:
Would probably better to use a proper mocking library. I believe CE already has `sinon` as a dependency

Reply:
I'm down with using a mocking library here too, unless there's an argument against it :)